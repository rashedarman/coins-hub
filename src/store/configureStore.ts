import { configureStore } from '@reduxjs/toolkit';
import cryptoSlice from './crypto/cryptoSlice';
import newsSlice from './news/newsSlice';

const rootReducer = {
  crypto: cryptoSlice,
  news: newsSlice,
};

const store = configureStore({
  reducer: rootReducer,
});

export default store;
