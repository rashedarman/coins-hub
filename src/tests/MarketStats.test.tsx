import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import MarketStats from '../components/MarketStats';
import store from '../store/configureStore';
import { coins } from './__mockData';

describe('Testing CoinsListTable component', () => {
  it('Renders CoinsListTable component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <MarketStats coin={coins[0]} />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
