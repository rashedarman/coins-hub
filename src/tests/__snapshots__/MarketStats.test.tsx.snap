// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CoinsListTable component Renders CoinsListTable component correctly 1`] = `
<div>
  <div
    class="mantine-1avyp1d"
  >
    <h3
      class="mantine-Text-root mantine-Title-root mantine-11dqeqv"
    >
      Market Stats
    </h3>
    <div
      class="mantine-Paper-root mantine-6h2oqv"
    >
      <div
        class="mantine-73o8aw"
      >
        <div
          class="mantine-1fr50if"
        >
          <div
            class="mantine-Text-root mantine-1m93l80"
          >
            Market Cap
          </div>
          <div
            class="mantine-Text-root mantine-76dy9l"
          >
            442.8B
          </div>
        </div>
        <div
          class="mantine-1fr50if"
        >
          <div
            class="mantine-Text-root mantine-1m93l80"
          >
            Fully Dilluted Valuation
          </div>
          <div
            class="mantine-Text-root mantine-76dy9l"
          >
            $
            482.1B
          </div>
        </div>
        <div
          class="mantine-1fr50if"
        >
          <div
            class="mantine-Text-root mantine-1m93l80"
          >
            Circulating Supply
          </div>
          <div
            class="mantine-Text-root mantine-76dy9l"
          >
            $
            19,285,012
          </div>
        </div>
        <div
          class="mantine-1fr50if"
        >
          <div
            class="mantine-Text-root mantine-1m93l80"
          >
            Total Supply
          </div>
          <div
            class="mantine-Text-root mantine-76dy9l"
          >
            $
            21000000
          </div>
        </div>
        <div
          class="mantine-1fr50if"
        >
          <div
            class="mantine-Text-root mantine-1m93l80"
          >
            Volume 24h
          </div>
          <div
            class="mantine-Text-root mantine-76dy9l"
          >
            $
            36.4B
          </div>
        </div>
      </div>
      <div
        class="mantine-Divider-root mantine-Divider-horizontal mantine-kqmy2u"
        role="separator"
      />
      <div
        class="mantine-1avyp1d"
      >
        <h4
          class="mantine-Text-root mantine-Title-root mantine-1gqz47j"
        >
          Bitcoin
           
          Price Update
        </h4>
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Bitcoin
           
          <b>
            price is
             
            $22,959.00
            ,
             
            down
             
            0.15
            %
             
          </b>
          in the last 24 hours, and the live market cap is
           
          <b>
            $
            442.8B
          </b>
          . It has circulating
           
          <b>
            supply volume of $
            19,285,012
             
            BTC
          </b>
           
          coins and a max. supply volume of
           
          21000000
           
          alongside $
          36.4B
           
          24h trading volume.
        </div>
      </div>
    </div>
  </div>
</div>
`;
